src/helpers.js

```diff
@@ -33,3 +36,17 @@ export function arraysEqual(a, b) {
 
   return a.every((element, index) => element === b[index]);
 }
+
+function memoizeString<T>(fn: (string) => T): (string) => T {
```

Comment:
I found this type signature really hard to parse. Is there a way to pull the `(string) => T` part into a Type?

This is invalid, but I'm wondering if anything exists similar to:

```js
type StringManipulator = (string) => T;

function memoizeString<T>(fn: StringManipulator<T>): StringManipulator<T> 
```